dbt_platform_helper/commands/environment.py

```diff
@@ -511,18 +444,29 @@ def remove_maintenance_page(session: boto3.Session, listener_arn: str):
     lb_client = session.client("elbv2")
 
     rules = lb_client.describe_rules(ListenerArn=listener_arn)["Rules"]
-    rules = lb_client.describe_tags(ResourceArns=[r["RuleArn"] for r in rules])["TagDescriptions"]
+    tag_descriptions = get_rules_tag_descriptions(rules, lb_client)
+    tag_descriptions = lb_client.describe_tags(ResourceArns=[r["RuleArn"] for r in rules])[
+        "TagDescriptions"
+    ]
 
     for name in ["MaintenancePage", "AllowedIps", "BypassIpFilter"]:
-        deleted = delete_listener_rule(rules, name, lb_client)
+        deleted = delete_listener_rule(tag_descriptions, name, lb_client)
 
         if name == "MaintenancePage" and not deleted:
             raise ListenerRuleNotFoundError()
 
 
-def get_public_ip():
-    response = requests.get("https://api.ipify.org")
-    return response.text
+def get_rules_tag_descriptions(rules: list, lb_client):
+    tag_descriptions = []
+    chunk_size = 20
+
+    for i in range(0, len(rules), chunk_size):
+        chunk = rules[i : i + chunk_size]
```

Comment:
This is [slice notation](https://www.geeksforgeeks.org/python-list-slicing/). So on first iteration it will be 0:20, then 20:40 on second and so on. It will iterate through the rules taking slices of max 20 elements at a time